bika/lims/browser/js/coffee/bika.lims.worksheet.coffee

```diff
@@ -0,0 +1,1070 @@
+     *
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_method_change °°°"
+    $el = $(event.currentTarget)
+
+    analysis_uid = $el.attr "uid"
+    method_uid = $el.val()
+
+    # Change the instruments to be shown for an analysis when the method selected changes
+    @load_analysis_method_constraint analysis_uid, method_uid
+
+
+  on_analysis_instrument_focus: (event) =>
+    ###
+     * Eventhandler when the instrument of an analysis is focused
+     *
+     * Only needed to remember the last value
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_analysis_instrument_focus °°°"
+
+    $el = $(event.currentTarget)
+    @previous_instrument = $el.val()
+    console.info @previous_instrument
+
+
+  on_analysis_instrument_change: (event) =>
+    ###
+     * Eventhandler when the instrument of an analysis changed
+     *
+     * If a new instrument is chosen for the analysis, disable this Instrument
+     * for the other analyses. Also, remove the restriction of previous
+     * Instrument of this analysis to be chosen in the other analyses.
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_analysis_instrument_change °°°"
+    $el = $(event.currentTarget)
+
+    analysis_uid = $el.attr "uid"
+    instrument_uid = $el.val()
+
+    # Disable New Instrument for rest of the analyses
+    $("table.bika-listing-table select.listing_select_entry[field='Instrument'][value!='#{instrument_uid}'] option[value='#{instrument_uid}']").prop "disabled", yes
+
+    # Enable previous Instrument everywhere
+    $("table.bika-listing-table select.listing_select_entry[field='Instrument'] option[value='#{@previous_instrument}']").prop "disabled", no
+
+    # Enable 'None' option as well.
+    $("table.bika-listing-table select.listing_select_entry[field='Instrument'] option[value='']").prop "disabled", no
+
+
+  on_detection_limit_change: (event) =>
+    ###
+     * Eventhandler when the detection limit changed
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_detection_limit_change °°°"
+    $el = $(event.currentTarget)
+
+    defdls = $el.closest("td").find("input[id^='DefaultDLS.']").first().val()
+    resfld = $el.closest("tr").find("input[name^='Result.']")[0]
+    uncfld = $el.closest("tr").find("input[name^='Uncertainty.']")
+
+    defdls = $.parseJSON(defdls)
+    $(resfld).prop "readonly", !defdls.manual
+
+    if $el.val() == "<"
+      $(resfld).val defdls['min']
+      # Inactivate uncertainty?
+      if uncfld.length > 0
+        $(uncfld).val ""
+        $(uncfld).prop "readonly", yes
+        $(uncfld).closest("td").children().hide()
+    else if $el.val() == ">"
+      $(resfld).val defdls["max"]
+      # Inactivate uncertainty?
+      if uncfld.length > 0
+        $(uncfld).val ""
+        $(uncfld).prop "readonly", yes
+        $(uncfld).closest("td").children().hide()
+    else
+      $(resfld).val ""
+      $(resfld).prop "readonly", no
+      # Activate uncertainty?
```

Comment:
Yes, uncertainty must be only active if the result is a "real" result, not a detection limit (it does not make any sense to calculate the uncertainty if the result introduced is a detection limit instead of a discrete result!) - see this comment https://github.com/senaite/senaite.core/pull/721/files#r174639383 for further details